### ISSUE TITLE
Enhance DataHarmonizer so that it can build a hierarchical pull-down menu based on LinkML dynamic enums

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/issues#160]
Currently, an enum like `EnvBroadScaleSoilEnum` in `src/nmdc_submission_schema/schema/nmdc_submission_schema.yaml` is built by [running an OAK command and then doing some semi-automated cleanup](https://github.com/microbiomedata/submission-schema/blob/592e9b644f0411ba195f44fd6027d4a75aef4c73/project.Makefile#L352-L363) on the output.

The resulting pull-down emphasizes the hierarchical nature of the permissible values (although the permissible values aren't sorted alphabetically within each rank)

![Screenshot from 2023-03-29 19-12-28](https://user-images.githubusercontent.com/25185273/228688319-fd523985-7380-47bb-9199-74425042f570.png)

**Comment by turbomam** [comment]
@cmungall has suggested that DataHarminzer would ideally have some baked-in sense of ontology and that this wouldn't be necessary.

Could we modify DataHormizer so that it builds the pull-down menus from [LinkML dynamic enumerations](https://linkml.io/linkml/schemas/enums.html#dynamic-enums) and displays seomthign that keeps this ontological look and feel without requiring that the PVs are indented with underline characters?

**Comment by mslarae13** [comment]
I think this relates to many other env issues. But I'm not sure I understand the specifics of this issue / what it wants to solve.

@turbomam is it that the enums just aren't sorted? Cuz that's in another issue
Or is it that we want like a widgit for ontology searching?

**Comment by ssarrafan** [comment]
@turbomam can you elaborate on this issues, if it's still important, and respond to Montana's question above?

**Comment by turbomam** [comment]
Yes, I was advocating for a pre-scoped ontology class browser widget. Presumably @pkalita-lbl would have to do the JavaScript coding, but I will help in any way I can.

For the mean time, I suggest that we stop the practice of making the underscore-based hierarchies. The code I wrote to do that was never very good. Maybe we could revisit it with the ontology access kit. 

_I_ still dislike the inflated size of the enumerations, due to multiple appearances of classes that have multiple paths to the root.

**Comment by mslarae13** [comment]
Will the new list of terms we curate in the env triad squad eliminate that inflated size? & the underscore-based hierarchies?
- Users did understand this in the user research. So how would you recommend we ensure that information isn't lost?
Ontology class browser widget, yes, but in the renewal. I don't think we have the time this FY.
